sparkpost.class.php

```diff
@@ -69,20 +77,20 @@ public function add_settings_link($links)
 
     public function set_from_name($name)
     {
-        if (!empty($this->options['from_name'])) {
-            return $this->options['from_name'];
-        } else {
-            return $name;
+        if (!empty($this->settings['from_name'])) {
+            $name = $this->settings['from_name'];
         }
+
+        return apply_filters('wpsp_sender_name', $name);
```

Comment:
thinking if we should really have these filters. it's just duplicating wordpress's own filters. if someone wants to set a different from, they should use `wp_mail_from` hook with different priorities.